src/service/articles.ts

```diff
@@ -0,0 +1,21 @@
+import { useQuery } from '@tanstack/react-query';
+import { PAGE_LIMIT } from '../constants';
+import { Article } from '../types/articles';
+
+const fetchArticles = async (
+  page: number,
+): Promise<{ articles: Article[]; articlesCount: number }> => {
+  const res = await fetch(
+    `${process.env.NEXT_PUBLIC_API_BASE_URL}articles?offset=${page}&limit=${PAGE_LIMIT}`,
+  );
+  const data = await res.json();
+  return data;
+};
+
+export const useGetAllArticlesQuery = (page: number) => {
+  const { data, ...rest } = useQuery(['posts', page], () =>
```

Comment:
key값도 바뀌면 좋겠네요! queryKey를 상수로 관리하는 것도 해보시면 좋을 것 같아요~

```suggestion
  const { data, ...rest } = useQuery(['articles', page], () =>
```